test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -337,16 +342,18 @@ public:
 
     flat_reader_assertions& produces(const schema& s, const mutation_fragment& mf) {
         testlog.trace("Expect {}", mutation_fragment::printer(s, mf));
+        if (mf.is_range_tombstone()) {
+            apply_rt(mf.as_range_tombstone());
```

Comment:
[ oops, this appears to have fallen into some sort of a black hole, my apologies! ]

I don't set `_check_rts` to `true` here because none of the tests that call this expect it (i.e. they'll break if you set `_check_rts` to `true` here).

That is, as far as I can see, existing tests that make use of the `flat_mutation_reader_assertions` class just so happen to split along the "actually want to check range tombstones" axis like this:
- those that invoke `produces_range_tombstone()` do want it
- those that invoke the various versions of the more generic `produces()` don't want it

I wish I had a more intelligent and/or systematic answer, but that's what I have, sorry...

Reply:
You mean that there are tests which call `produces(range_tombstone([a, b]))`, and the reader doesn't actually produce range tombstones for the range [a b]? That seems like a bug in the test or in the code under test. 

Can you give an example of a test which fails?

I think produces() should check that the range tombstone actually occurs.

Reply:
ping @cmm

Reply:
well, the only problematic test (which is runs on many mutations, though) is "test_flat_mutation_reader_consume_single_partition", and the problem with it is not that it doesn't check that a range tombstone occurs, but that more tombstones are emitted in the partition past the point where it stops caring.

the error (if I add `_check_rts = true`) looks like this:
```
./test/lib/flat_mutation_reader_assertions.hh(59): fatal error: in "test_flat_mutation_reader_consume_single_partition": Expected {{range_tombstone: start={bound: prefix=ckp{0080030000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000806d02000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000}, kind=incl start}, end={bound: prefix=ckp{00800301000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000}, kind=excl end}, {tombstone: timestamp=-9223372036854775797, deletion_time=1}}}, but got {{range_tombstone: start={bound: prefix=ckp{0080030000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000806d02000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000}, kind=incl start}, end={bound: prefix=ckp{00800301000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000}, kind=excl end}, {tombstone: timestamp=-9223372036854775797, deletion_time=1}}, {range_tombstone: start={bound: prefix=ckp{00800301000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000}, kind=incl start}, end={bound: prefix=ckp{0080e20200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000802a03000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000}, kind=incl end}, {tombstone: timestamp=-9223372036854775795, deletion_time=2}}}
```